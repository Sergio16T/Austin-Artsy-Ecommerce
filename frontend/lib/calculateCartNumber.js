export default function(cart) {
    return cart.reduce((startValue, element)=> {
        return startValue + element.quantity; 
    }, 0); 
}