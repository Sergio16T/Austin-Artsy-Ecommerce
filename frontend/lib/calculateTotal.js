export default function(cart) {
    return cart.reduce((startValue, element) => {
    if(!element.item) return startValue; 
    return startValue + (element.quantity * element.item.price)}, 0); 
    
}