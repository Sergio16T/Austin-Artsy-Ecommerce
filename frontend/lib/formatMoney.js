export default function(amount) {
    let price = amount / 100; 
    return new Intl.NumberFormat('en', {
        style: "currency",
        currency: "USD",
    }).format(price);
}

/* when receiving money from form make sure it always has two floating point numbers and then just multiply it by 100
to store it in graphql as cents in int type then when I'm displaying money multiply it by 100.
that way I can format it for use later.. so it's always going to be handled in cents 
165.33 * 100
16533
165 * 100
16500
16500/100
165
16533/100
165.33
 */ 
