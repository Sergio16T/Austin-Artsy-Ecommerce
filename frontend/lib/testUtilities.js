import casual from 'casual'; 
//.seed() is to provide ability to use a specific seed in order to get a repeatable random sequence for our testing purposes
casual.seed(111); 

const mockItem = () => ({
    // __typename: 'Item',
    id: '123',
    price: 5000,
    user: null,
    image: 'dog.jpg',
    title: 'dogs are the best',
    description: 'dogs description',
    largeImage: 'dog-large.jpg',
}); 

const mockUserWithItemsInCart = () =>  ({
    __typename: 'User', 
    id: "123", 
    name: casual.name, 
    email: casual.email, 
    permissions: ["ADMIN"], 
    cart: [{
        id: 'abc', 
        quantity: 2, 
        item: {
            id: 'dgh', 
            price: 15000, 
            image: 'www.dog.image.com',
            title: "puppy"
        }
        },
    ]
}); 
const mockUser = () => ({
    __typename: 'User', 
    id: "123", 
    name: casual.name, 
    email: casual.email, 
    permissions: ['ADMIN'], 
    cart: [],
    orders: []
}); 
const mockOrderItem = () => ({
    __typename: 'OrderItem',
    id: casual.uuid,
    image: `${casual.word}.jpg`,
    title: casual.words(),
    price: 4234,
    quantity: 1,
    description: casual.words(),
  });
  
  const mockOrder = () => ({
    __typename: 'Order',
    id: 'ord123',
    charge: 'ch_123',
    total: 40000,
    items: [mockOrderItem(), mockOrderItem()],
    createdAt: '2018-04 - 06T19: 24: 16.000Z',
    user: mockUser(),
  });
  
  const mockCartItem = overrides => ({
    __typename: 'CartItem',
    id: 'omg123',
    quantity: 3,
    item: mockItem(),
    user: mockUser(),
    ...overrides,
  });
//mock LocalStorage 
class MockLocalStorage {
    constructor(){
        this.store={}; 
    }

    clear() {
        this.store = {}; 
    }
    getItem(key) {
        return this.store[key] || null; 
    }
    setItem(key, value) {
        this.store[key] = value.toString(); 
    }
    removeItem(key) {
        delete this.store[key]; 
    }
}

export { 
    mockItem, 
    MockLocalStorage, 
    mockUser, 
    mockCartItem, 
    mockOrderItem, 
    mockOrder }; 