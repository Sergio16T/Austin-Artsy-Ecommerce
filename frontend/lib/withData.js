import withApollo from 'next-with-apollo'; 
import ApolloClient from 'apollo-boost'; 
import { endpoint } from '../config'; 

function createClient({ headers}) {
    return new ApolloClient({
        uri: process.env.NODE_ENV === 'development' ? endpoint : endpoint, 
        // on each request pass the headers and setContext change credentials to 'include' for whether the user agent should send cookies from other domains in case of CORS
        request: operation => {
            operation.setContext({
                fetchOptions: {
                    credentials: 'include', // logged in cookies present ? include.  (default value is same-origin) 
                },
                headers, 
            }); 
        },
    }); 
}

export default withApollo(createClient); 