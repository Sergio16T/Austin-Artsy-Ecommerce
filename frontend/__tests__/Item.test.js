import ItemComponent, { ItemButtons } from '../components/Item'; 
import { render, screen } from '@testing-library/react'; 
import { MockedProvider } from '@apollo/react-testing';
import { mockItem } from '../lib/testUtilities'; 

const item = mockItem();

const mocks = {};

describe('<Item/>', () => {
    it('renders and matches the snapshot', () => {
      const { container } = render(
        <MockedProvider>
          <ItemComponent item={item} />
        </MockedProvider>
      );
      expect(container).toMatchSnapshot();
    });

    it('renders the image', () => {
        const { container } = render(
            <MockedProvider>
              <ItemComponent item={item} />
            </MockedProvider>
          );
          const img = screen.getByAltText(item.title)
          expect(img).toBeInTheDocument; 
    });
    it('renders the priceTag and title', () => {
        const { container, debug } = render(
            <MockedProvider>
              <ItemComponent item={item} />
            </MockedProvider>
          );
          expect(screen.getByText("$50.00")).toBeInTheDocument(); 
          const link = container.querySelector('a'); 
          expect(link).toHaveTextContent(item.title); 
          expect(link).toHaveAttribute('href', '/item?id=123');
    });  

}); 

