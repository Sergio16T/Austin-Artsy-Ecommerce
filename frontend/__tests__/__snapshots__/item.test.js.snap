// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<div>
  <div
    class="ItemStyles__StyledItem-sc-2ja5ol-1 dXvgRY"
  >
    <img
      alt="dogs are the best"
      src="dog.jpg"
    />
    <div
      class="Item__Title-gl8sfg-0 huNLuC"
    >
      <a
        href="/item?id=123"
      >
        dogs are the best
      </a>
    </div>
    <p>
      dogs description
    </p>
    <p>
      $50.00
    </p>
    <div
      class="Item__ItemButtons-gl8sfg-1 fydofk"
    >
      <a
        href="/update?id=123"
        id="editButton"
      >
         Edit 
      </a>
      <button
        type="button"
      >
        Add to Cart
      </button>
      <button>
        Delete
      </button>
    </div>
  </div>
</div>
`;
